test/base_option_3_policy_test.rb

```diff
@@ -0,0 +1,210 @@
+require "test_helper"
+require "accessly/policy/base_option_3"
+
+describe Accessly::Policy::BaseOption3 do
+############
+# NOTE TO SELF
+# BaseOption3 implements an API like this:
+# UserPolicy.new(user).can?(view)
+# UserPolicy.new(user).can?(:view, other_user)
+# UserPolicy.new(user).list(:view)
+#
+# BaseOption2 implements it the same way.
+# However BaseOption3 allows us to more easily override
+# permission checks than BaseOption2.
+#
+# We will need to decide which one to move forward with.
+############
+
+  class UserPolicyOption3 < Accessly::Policy::BaseOption3
+
+    actions(
+      view: 1,
+      edit_basic_info: 2,
+      change_role: 3,
+      destroy: 4
+    )
+
+    actions_on_objects(
+      view: 1,
+      edit_basic_info: 2,
+      change_role: 3,
+      email: 4
+    )
+
+    def self.object_type
+      User
+    end
+  end
+
+  class CustomizedPolicyOption3 < UserPolicyOption3
+    # Customize a general action check
+    override_action_check :destroy, ->(actor) { true if actor.name == "Aaron" }
```

Comment:
For each permission check, the logic is like this:

- Is there an override for this action?
    - If yes, run it
    - Is the result of that overridden check `nil`?
        - If so, do the normal `Accessly::Query` lookup
- If not
    - Do the normal `Accessly::Query` lookup

Reply:
I like this API - but is our main use case to override on the given 'actor'.  So for any given action, if the actor is an "admin" return true 

Also - How do you envision the policies handle the multiple actor scenario? Would a policy accept am actor and then know how to get the actor relationships?